src/createTheme.ts

```diff
@@ -0,0 +1,193 @@
+import { DefaultTheme } from 'styled-components';
+import { ThemeVariants } from './types/ThemeVariants';
+import { defaultThemeColors } from './defaultThemeColors';
+import { ThemeBreakpoints } from './types/ThemeBreakpoints';
+import { ButtonVariants } from './types/ButtonVariants';
+import { BadgeVariants } from './types/BadgeVariants';
+import { defaultThemeSizes } from './defaultThemeSizes';
+
+export const createTheme = (customTheme?: DefaultTheme): DefaultTheme => {
```

Comment:
originally from src/theme.js, converted to ts and gave it a better name and took some contants out to make the file shorter.